public/app/plugins/datasource/azuremonitor/azure_monitor/azure_monitor_datasource.ts

```diff
@@ -293,12 +288,12 @@ export default class AzureMonitorDatasource extends DataSourceWithBackend<AzureM
       apiVersion,
       // Only use the first query, as the metric metadata should be the same for all queries
       this.replaceSingleTemplateVariables(query),
-      this.templateSrv,
+      getTemplateSrv(),
       multipleResources,
       region
     );
     return this.getResource(url).then((result: AzureMonitorMetricsMetadataResponse) => {
-      return ResponseParser.parseMetadata(result, this.templateSrv.replace(metricName));
+      return ResponseParser.parseMetadata(result, getTemplateSrv().replace(metricName));
```

Comment:
Does it make sense to retrieve the `templateSrv` multiple times rather than having it on the class?

Reply:
I don't think it's that big of a deal. When templateSrv is a property, the data source needs to be remade for every test. Maybe this isn't a big deal. What do you think?

Reply:
It feels cleaner to have it at the class level rather than retrieving a new one everytime we need it. I don't think it's a big deal remaking the data source for every test because each test should be in isolation maybe?